File: BACKEND/FragranceWebshop_Entities/Entity Models/Perfum.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FragranceWebshop_Entities.Entity_Models
{
    public class Perfum
    {
        public string PerfumId { get; set; }
        public string PerfumName { get; set; }
        public string RecommendedSeason { get; set; }
        public int Price { get; set; }

        public string ImageUrl { get; set; }

        public virtual ICollection<Purchase>? Purchases { get; set; }

        public Perfum(string perfumName, string recommendedSeason, int price, string imageUrl)
        {
            PerfumId = Guid.NewGuid().ToString();
            PerfumName = perfumName;
            RecommendedSeason = recommendedSeason;
            Price = price;
            ImageUrl = imageUrl;

        }

        public Perfum()
        {
            
        }
    }
}

File: FRONTEND/src/app/perfum.ts
export class Perfum {

    perfumId:string = ""
    perfumName:string = "";
    recommendedSeason:string = "";
    price:number = 0;
    purchaseCount:number = 0;
    imageUrl:string = ""

    constructor(perfumId:string, perfumName:string, recommendedSeason:string, price:number, purchaseCount:number, imageUrl:string)
    {
        this.perfumId = perfumId;
        this.perfumName = perfumName;
        this.recommendedSeason = recommendedSeason;
        this.price = price;
        this.purchaseCount = purchaseCount;
        this.imageUrl = imageUrl;

    }


}
